lib/units/websocket/index.js

```diff
@@ -334,6 +336,38 @@ module.exports = function(options) {
         .on('user.settings.reset', function() {
           dbapi.resetUserSettings(user.email)
         })
+        .on('user.keys.accessToken.generate', function(data) {
+          var expiry = Date.now() + 100 * 365 * 24 * 3600
+            , jwt = jwtutil.encode({
+                payload: {
+                  email: user.email
+                , name: user.name
+                }
+              , secret: options.secret
+              , expiry: expiry
+            })
+
+          var tokenId = uuid.v4()
+            , title = data.title
+
+          return dbapi.saveUserAccessToken(user.email, {
```

Comment:
Doesn't look like there's an example of the token being used yet, but if we were able to use the JWT token as the API key directly (or perhaps re-encode the whole thing once more with base64 if the dots look sketchy), we wouldn't need to access the database at all to authenticate the request. However I'm not sure how we could then have multiple access tokens per user without embedding some random data in the token, making it longer.

Also, `node-uuid` seems to use `crypto.randomBytes()` behind the scenes, and it's v4 so it should be OK, but UUIDs are not ideal as access tokens. There's some discussion over at https://news.ycombinator.com/item?id=10631806. I think a sha256 of some (>128) random bytes from `crypto.randomBytes()` would be better and it'd look a bit more like a normal access token.